### PR TITLE
[Berlin-WDFT-MAR2021] Michelle Liebheit

### PR DESCRIPTION
**Pull request by miliberlin**
I haven't finished the last bonus task yet.

### CODE REVIEW

File: src/movies.js [hunk 1]
```diff
@@ -1,16 +1,133 @@
...
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getAllUniqueDirectors(movies) {
+    return [...new Set(getAllDirectors(movies))];
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    let spielbergMovies = movies.filter(function(movie) {
```
Comment (wax-bear) [inline src/movies.js:20]: Will this variable be reassigned? If not, how about using a const?

File: src/movies.js [hunk 2]
```diff
@@ -1,16 +1,133 @@
...
+        if (movie['director'] === 'Steven Spielberg' && movie["genre"].includes('Drama')) return true
+    })
+    return spielbergMovies.length;
+
```
Comment (wax-bear) [inline src/movies.js:24]: How about placing this empty line above the return statement?

File: src/movies.js [hunk 3]
```diff
@@ -1,16 +1,133 @@
...
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies) {
+    if (movies.length === 0) return 0;
+    let avg = movies.reduce(function(acc, movie) {
+        if (movie.hasOwnProperty('rate') && movie['rate'] !== '') {
+            return acc + movie['rate']
```
Comment (wax-bear) [inline src/movies.js:33]: Please be mindful about the use of semicolons.

File: src/movies.js [hunk 4]
```diff
@@ -1,16 +1,133 @@
...
 
+function getAllUniqueDirectors(movies) {
+    return [...new Set(getAllDirectors(movies))];
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    let spielbergMovies = movies.filter(function(movie) {
+        if (movie['director'] === 'Steven Spielberg' && movie["genre"].includes('Drama')) return true
```
Comment (wax-bear) [inline src/movies.js:21]: Instead of returning true, the condition can be returned.

File: src/movies.js [hunk 5]
```diff
@@ -1,16 +1,133 @@
...
+        return acc + 0
+    },0)
+    return Number((avg / movies.length).toFixed(2))
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(movies) {
+    let dramaMovies = movies.filter(function(movie) {
+        if (movie['genre'].includes('Drama')) {
+            return true
```
Comment (wax-bear) [inline src/movies.js:44]: Instead of returning true, the condition can be returned.

**Review by wax-bear — COMMENTED** [review]
Great stuff Michelle!
I placed some comments for potential improvements. Also, in general, more empty lines/vertical formatting can be used to make the code more readable.
If you have a question, please reach out.

File: src/movies.js (hunk 4 above)
Comment (miliberlin) [reply src/movies.js:21]: Would I just write it like `return if ...`?

Comment (wax-bear) [reply src/movies.js:21]: You can simply write:
`return movie['director'] === 'Steven Spielberg' && movie["genre"].includes('Drama');`

**Pull request closed without merging**